src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs

```diff
@@ -244,21 +264,100 @@ public Task<IList<BlobDescriptor>> ListBlobsAsync(string containerName)
+                {
+                    await UpdateBlobPropertiesAsync(containerName, blobName, properties);
+                }
             }
             catch (Exception ex)
             {
                 throw ex.ToStorageException();
             }
         }
-
+        
         public void UpdateBlobProperties(string containerName, string blobName, BlobProperties properties)
         {
-            return;
+            UpdateBlobPropertiesAsync(containerName, blobName, properties).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        public async Task UpdateBlobPropertiesAsync(string containerName, string blobName, BlobProperties properties)
+        {
+            try
+            {
+                var path = ExtractFullPathAndProtectAgainstPathTraversal(containerName, blobName);
+                var metaPath = CreateMetadataPath(path);
+
+                var json = JsonSerializer.Serialize(properties, _jsonSerialiserOptions);
+
+                using (var file = File.Create(metaPath))
+                using (var streamWriter = new StreamWriter(file))
+                {
+                    await streamWriter.WriteAsync(json);
+                    await streamWriter.FlushAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw ex.ToStorageException();
+            }
+        }
+        
+        private string ExtractFullPathAndProtectAgainstPathTraversal(string containerName, string blobName)
+        {
+            var dir = Path.Combine(_basePath, containerName);
+            var path = Path.Combine(dir, blobName);
+
+            if (!Path.GetFullPath(path).StartsWith(dir, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new UnauthorizedAccessException("Detected path traversal attempt.").ToStorageException();
+            }
+
+            return path;
+        }
+        
+        
+        private static string CreateMetadataPath(string path)
+        {
+            return $"{path}-meta.json";
+        }
+        
+        private void MergeDescriptorWithCustomMetaIfExists(string fullPathToBlob, BlobDescriptor descriptor)
+        {
+            var meta = GetFileMetaIfExists(fullPathToBlob);
+            if (meta != default)
+            {
+                if (!string.IsNullOrWhiteSpace(meta.ContentDisposition))
+                {
+                    descriptor.ContentDisposition = meta.ContentDisposition;
+                }
+
+                if (!string.IsNullOrWhiteSpace(meta.ContentType))
+                {
+                    descriptor.ContentType = meta.ContentType;
+                }
+
+                descriptor.Security = meta.Security;
+                descriptor.Metadata = meta.Metadata;
+            }
+        }
+
+        private BlobDescriptor GetFileMetaIfExists(string fullPathToBlob)
+        {
+            return GetFileMetaIfExistsAsync(fullPathToBlob).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Lets create a non-async version of GetFileMetaIfExists, so we don't have to use `GetResult()`. `StreamReader` has a `ReadToEnd()` method, to replace the async one.